pkg/transport/transport.go

```diff
@@ -15,43 +15,80 @@
 package transport
 
 import (
+	"context"
 	"net"
 	"net/http"
 	"strings"
 	"time"
 )
 
+type CancelableTransport struct {
+	*http.Transport
+	ctx    context.Context
+	cancel context.CancelFunc
+}
+
+func (c *CancelableTransport) Ctx() context.Context { return c.ctx }
+func (c *CancelableTransport) Cancel()              { c.cancel() }
+
+func (c *CancelableTransport) RoundTrip(req *http.Request) (*http.Response, error) {
+	if req.Context() != context.Background() {
```

Comment:
If the context of the request is overridden with TODO, then that request can't be cancelled even if `CancelableTransport.Cancel()` is called right?  If yes, is that an expected behavior?